src/lvl_script.c

```diff
@@ -4119,6 +4214,162 @@ TbBool script_level_up_creature(PlayerNumber plyr_idx, long crmodel, long criter
     return true;
 }
 
+/**
+ * Cast a spell on a creature which meets given criteria.
+ * @param plyr_idx The player whose creature will be affected.
+ * @param crmodel Model of the creature to find.
+ * @param criteria Criteria, from CreatureSelectCriteria enumeration.
+ * @param magic_caster_splvl encoded bytes: magic id, caster player index and spell lvl (MSD order)
+ * @return True if a creature was found and the spell was succesfully cast.
+ */
+TbBool script_cast_spell_on_creature(PlayerNumber plyr_idx, long crmodel, long criteria, long magic_caster_splvl)
+{
+    struct Thing *thing;
+    switch (criteria)
+    {
+    case CSelCrit_Any:
+        thing = get_random_players_creature_of_model(plyr_idx, crmodel);
+        break;
+    case CSelCrit_MostExperienced:
+        thing = find_players_highest_level_creature_of_breed_and_gui_job(crmodel, CrGUIJob_Any, plyr_idx, 0);
+        break;
+    case CSelCrit_MostExpWandering:
+        thing = find_players_highest_level_creature_of_breed_and_gui_job(crmodel, CrGUIJob_Wandering, plyr_idx, 0);
+        break;
+    case CSelCrit_MostExpWorking:
+        thing = find_players_highest_level_creature_of_breed_and_gui_job(crmodel, CrGUIJob_Working, plyr_idx, 0);
+        break;
+    case CSelCrit_MostExpFighting:
+        thing = find_players_highest_level_creature_of_breed_and_gui_job(crmodel, CrGUIJob_Fighting, plyr_idx, 0);
+        break;
```

Comment:
It would be good if you extract this into a function like
`struct Thing *script_get_creature_by_criteria(PlayerNumber plyr_idx, long criteria);`